runtime/manifest.js

```diff
@@ -219,6 +230,40 @@ class Manifest {
           connection.connectToView(targetView);
         }
       }
+      for (let slotConnectionItem of item.slotConnections) {
+        let slotConn = particle.addSlotConnection(slotConnectionItem.param, slotConnectionItem.dir);
+        if (!slotConnectionItem.name) {
+          continue;
+        }
+        assert(items.byName.has(slotConnectionItem.name));
+        let {item, slot} = items.byName.get(slotConnectionItem.name);
+        slotConn.connectToSlot(slot);
+        if (item.ref.view || (item.ref.tags.length > 0)) {
+          let foundConnection = false;
+          for (let particleViewConn of Object.values(particle.connections)) {
+            if (!particleViewConn.view) continue;
+            if ((particleViewConn.view.localName == item.ref.view) ||
+                (item.ref.tags.every(t => particleViewConn.view.tags.indexOf(t) >= 0))) {
+              slotConn.connectToView(particleViewConn.name);
+              foundConnection = true;
+              break;
+            }
+          }
+          // add new unnamed connections
+          if (!foundConnection) {
+            let viewConn = particle.addUnnamedConnection();
```

Comment:
> Or that can only be implied by having a particle consume it and no other particle in the recipe providing it? (which might lead to the strategizer finding other ways to create that slot that might not be intended? probably fine in most cases)

I hadn't considered consumes at the recipe level, but I don't think we need it. What you describe here seems reasonable to me.